### ISSUE TITLE
Binary add op BF16 has lower performance than FP32

### ISSUE THREAD

**Issue by Leslie-Fang** [tensorflow/tensorflow#37975]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow):  yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04):  Centos 7.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: 
- TensorFlow installed from (source or
binary): - TensorFlow version (use command below):  binary pip install tensorflow==2.1.0
- Python version: - Bazel
version (if compiling from source):
- GCC/Compiler version (if compiling from
source): 
- CUDA/cuDNN version: - GPU model and memory:

**Describe the current behavior**
Running the binary op of two tensor, change the tensor type to Bfloat16 increase  the running time of 2x

**Describe the expected behavior**
Bfloat16 has lower memory consumption should be a little bit quick.

**Standalone code to reproduce the issue** 
pip install tensorflow==2.1.0
```
import tensorflow as tf
tf.compat.v1.disable_eager_execution()

if __name__ == "__main__":
  input_shape = [1, 224, 224, 1024]
  images = tf.random.uniform(input_shape, 0.0, 255.0, dtype=tf.float32, name='images')
  images2 = tf.random.uniform(input_shape, 0.0, 255.0, dtype=tf.float32, name='images2')
  images_bf16 = tf.random.uniform(input_shape, 0.0, 255.0, dtype=tf.bfloat16, name='images_bf16')
  images_bf16_2 = tf.random.uniform(input_shape, 0.0, 255.0, dtype=tf.bfloat16, name='images_bf16_2')
  mysum = tf.add(images, images2)
  mysum_bf16 = tf.add(images_bf16, images_bf16_2)
  with tf.compat.v1.Session() as sess:
    def run():
      res = sess.run(mysum)
      #print(res)
    def run2():
      res = sess.run(mysum_bf16)
    import timeit
    cost = timeit.timeit(stmt=run, number=200)
    print("fp32 cost time: {}".format(cost))
    cost_bf16 = timeit.timeit(stmt=run2, number=200)
    print("bf16 cost time: {}".format(cost_bf16))

```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

Currently result:
fp32 cost time: 15.485223675030284
bf16 cost time: 30.486599242896773

**Comment by Saduf2019** [comment]
i am able to replicate the issue, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/682a5774dd443e218ffbe86574dcdcd1/37975.ipynb)

**Comment by rmothukuru** [comment]
@Leslie-Fang,
Sorry for the delayed response. Can you please us [Mixed Precision](https://www.tensorflow.org/guide/mixed_precision) and let us know if it helps? Thanks!

**Comment by tilakrayal** [comment]
Was able to reproduce the issue in tf v2.7.Please find the gist [here](https://colab.sandbox.google.com/gist/tilakrayal/78b6611062cc3d87c24a382af2902456/untitled160.ipynb).

**Comment by mohantym** [comment]
Hi @Leslie-Fang ! This issue is getting resolved in the  2.8 version. Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/a78fe100d2ced4df36c31dbf7d2c03dc/untitled160.ipynb#scrollTo=oggjNRN6vJcQ) for reference. Thanks!
